app/workers/campaign_finisher_worker.rb

```diff
@@ -1,10 +1,8 @@
-class CampaignFinisherWorker
+class CampaignFinisherWorker < ProjectBaseWorker
   include Sidekiq::Worker
   sidekiq_options retry: false
 
-  def perform project_id
-    resource = Project.find project_id
-    Rails.logger.info "[FINISHING PROJECT #{resource.id}] #{resource.name}"
-    resource.finish
+  def perform id
+    resource_action id, :finish
```

Comment:
And here we can use as:
```ruby
def perform id
  resource(id).finish
end
```
we can avoid the send and make the code more readable